libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1681,352 +1681,556 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
 	}
 }
 
-bool TypeChecker::visit(FunctionCall const& _functionCall)
+TypePointer TypeChecker::typeCheckTypeConversionAndRetrieveReturnType(
+	FunctionCall const& _functionCall,
+	TypePointer const _expressionActualType
```

Comment:
```suggestion
	TypePointer const _targetType
```
("actual type" is not the right name in this context, I would argue)

Reply:
Or you can also just call it `_resultType` and make it non-const.

Reply:
As mentioned in the comment above, _expressionActualType parameter is now gone, typeCheckTypeConversionAndRetrieveReturnType() extracts the expression result type itself now and it is assigned to 'resultType' which is certainly a better name than 'actual type'.